server/ServerNetworking.cpp

```diff
@@ -207,7 +207,17 @@ void PlayerConnection::SendMessage(const Message& message) {
         ErrorLogger(network) << "PlayerConnection::SendMessage can't send message when not transmit connected";
         return;
     }
-    m_service.post(boost::bind(&PlayerConnection::SendMessageImpl, shared_from_this(), message));
+    m_service.post(boost::bind(&PlayerConnection::SendMessageImpl, shared_from_this(), message, std::function<void(bool)>()));
+}
+
+void PlayerConnection::SendMessage(const Message& message, std::function<void(bool)> callback) {
+    if (!m_valid) {
+        ErrorLogger(network) << "PlayerConnection::SendMessage can't send message when not transmit connected";
+        if (callback)
+            callback(false);
+        return;
+    }
+    m_service.post(boost::bind(&PlayerConnection::SendMessageImpl, shared_from_this(), message, callback));
```

Comment:
Or what is the design here? Send a message with a callback function then store that? Seems like they should be separate functions... or add a signal that is invoked whenever a message is sent and other code can connect to it...

Reply:
Yes, next I'll attach update of that empire property to that signal after sending turn message.

Reply:
my point is that you don't need to pass in a callback and store it... rather have a signal that's always present that fires whenever a message is sent. then wherever you want to be notified about that happening, connect that signal to something.

Reply:
You mean `boost::signals2::signal` ?

Reply:
I've looked `EventSignal` usage but not sure how to use it.

Reply:
Yes  with a boost signal. There are various uses in the UI and gamestate code. eg. For signalling that a UI widget value changed ir a button was clicked or that a UniverseObject's state changed.